worker/gouging.go

```diff
@@ -126,74 +135,118 @@ func checkUploadGouging(gs api.GougingSettings, hs rhpv2.HostSettings, minShards
-	if overflow {
-		return fmt.Errorf("overflow detected when multiplying %v * %v in upload gouging", upptb, totalShards)
+
+	// check max collateral
+	if hs.MaxCollateral.IsZero() {
+		return errors.New("MaxCollateral of host is 0")
 	}
-	uploadPrice := uploadPriceTotalShards.Div64(uint64(minShards))
-	if !gs.MaxUploadPrice.IsZero() && uploadPrice.Cmp(gs.MaxUploadPrice) > 0 {
-		return fmt.Errorf("cost per TiB exceeds max ul price: %v>%v", uploadPrice, gs.MaxUploadPrice)
+	if hs.MaxCollateral.Cmp(gs.MinMaxCollateral) < 0 {
+		return fmt.Errorf("MaxCollateral is below minimum: %v<%v", hs.MaxCollateral, gs.MinMaxCollateral)
 	}
 
 	return nil
 }
 
-func checkFormContractGouging(gs api.GougingSettings, hs rhpv2.HostSettings) error {
+func checkPriceGougingPT(gs api.GougingSettings, pt *rhpv3.HostPriceTable) error {
 	// check base rpc price
-	if !gs.MaxRPCPrice.IsZero() && hs.BaseRPCPrice.Cmp(gs.MaxRPCPrice) > 0 {
-		return fmt.Errorf("rpc price exceeds max: %v>%v", hs.BaseRPCPrice, gs.MaxRPCPrice)
+	if !gs.MaxRPCPrice.IsZero() && gs.MaxRPCPrice.Cmp(pt.InitBaseCost) < 0 {
+		return fmt.Errorf("init base cost exceeds max: %v>%v", pt.InitBaseCost, gs.MaxRPCPrice)
+	}
+
+	// check ReadLengthCost - should be 1H as it's unused by hosts
+	if types.NewCurrency64(1).Cmp(pt.ReadLengthCost) < 0 {
+		return fmt.Errorf("ReadLengthCost of host is %v but should be %v", pt.ReadLengthCost, types.NewCurrency64(1))
+	}
+
+	// check WriteLengthCost - should be 1H as it's unused by hosts
+	if types.NewCurrency64(1).Cmp(pt.WriteLengthCost) < 0 {
+		return fmt.Errorf("WriteLengthCost of %v exceeds 1H", pt.WriteLengthCost)
+	}
+
+	// check MemoryTimeCost - should be 1H as it's unused by hosts
+	if types.NewCurrency64(1).Cmp(pt.MemoryTimeCost) < 0 {
+		return fmt.Errorf("MemoryTimeCost of %v exceeds 1H", pt.WriteLengthCost)
 	}
 
 	// check contract price
-	if !gs.MaxContractPrice.IsZero() && hs.ContractPrice.Cmp(gs.MaxContractPrice) > 0 {
-		return fmt.Errorf("contract price exceeds max: %v>%v", hs.ContractPrice, gs.MaxContractPrice)
+	if !gs.MaxContractPrice.IsZero() && pt.ContractPrice.Cmp(gs.MaxContractPrice) > 0 {
+		return fmt.Errorf("contract price exceeds max: %v>%v", pt.ContractPrice, gs.MaxContractPrice)
 	}
 
-	// check MaxCollateral
-	if hs.MaxCollateral.IsZero() {
+	// check max storage
+	if !gs.MaxStoragePrice.IsZero() && pt.WriteStoreCost.Cmp(gs.MaxStoragePrice) > 0 {
+		return fmt.Errorf("storage price exceeds max: %v>%v", pt.WriteStoreCost, gs.MaxStoragePrice)
+	}
+
+	// check max collateral
+	if pt.MaxCollateral.IsZero() {
 		return errors.New("MaxCollateral of host is 0")
 	}
-	if hs.MaxCollateral.Cmp(gs.MinMaxCollateral) < 0 {
-		return fmt.Errorf("MaxCollateral is below minimum: %v<%v", hs.MaxCollateral, gs.MinMaxCollateral)
+	if pt.MaxCollateral.Cmp(gs.MinMaxCollateral) < 0 {
+		return fmt.Errorf("MaxCollateral is below minimum: %v<%v", pt.MaxCollateral, gs.MinMaxCollateral)
 	}
+
 	return nil
```

Comment:
Ok so no idea why GH decided to get rid of my earlier comment but here we got again. 
A list of fields that I think are not checked right now and probably should be while we are at it.

HostBlockHeight - probably needs to be checked somewhere but not sure if this is the place
UpdatePriceTableCost - might always be 1H?
AccountBalanceCost- might always be 1H?
FundAccountCost - might always be 1H?
LatestRevisionCost - might always be 1H?
SubscriptionMemoryCost - maybe leave a TODO
SubscriptionNotificationCost - maybe leave a TODO
DropSectorsBaseCost - might always be 1H?
DropSectorsUnitCost - might always be 1H?
HasSectorBaseCost - might always be 1H?
RenewContractCost - might always be 1H?
RevisionBaseCost - might always be 1H?
SwapSectorBaseCost - might always be 1H?
TxnFeeMinRecommended - should probably check this
TxnFeeMaxRecommended - should probably check this
MaxDuration - should probably include this in scoring the same way we treat the sector equivalent
WindowSize - should be checked although we might not be doing that for the settings either right now